fem/nonlinearform.cpp

```diff
@@ -109,13 +134,25 @@ const Vector &NonlinearForm::Prolongate(const Vector &x) const
 
 void NonlinearForm::Mult(const Vector &x, Vector &y) const
 {
+   const Vector &px = Prolongate(x);
+   if (P) { aux2.SetSize(P->Height()); }
+
+   // If we are in parallel, ParNonLinearForm::Mult uses the aux2 vector.
+   // In serial, place the result directly in y.
+   Vector &py = P ? aux2 : y;
+
+   if (ext)
+   {
+      ext->Mult(px, py);
+      py.HostRead();
+      return;
+   }
```

Comment:
I'm curious if this is the correct operation here. I always thought this function was setup to form the RHS side of our equations or at least that's how I've always used it in my code. Based on the ```BilinearForm``` of how PA is setup, I would have imagined that actions (```Mult``` and ```MultTranspose```) of ```ext``` would would have been used in the ```NonlinearForm::GetGradient``` operator. It's entirely possible that I'm missing something from the translation of what's being down in the ```BilinearForm``` and the ```NonlinearForm``` form for the PA formulations.